components/Login/index.tsx

```diff
@@ -0,0 +1,63 @@
+import Link from 'next/link'
+import Portal from '../Portal'
+import { useRouter } from 'next/router'
+import * as S from './style'
+import { useEffect } from 'react'
+import { gauthLoginUri } from '@/lib/GauthLoginUrI'
+import * as SVG from '@/assets/svg'
+import { accessExp, accessToken, refreshExp, refreshToken } from '@/lib/token'
+import { useFetch } from '@/hooks'
+
+export default function Login() {
+  const router = useRouter()
+  const gauthCode = router.query.code?.toString()
+
+  type TokensType = {
+    accessToken: string
+    refreshToken: string
+    accessExp: string
+    refreshExp: string
+  }
+
+  const { fetch } = useFetch({
+    url: 'auth',
+    method: 'post',
+    onSuccess: (data: TokensType) => {
+      if (!data) return
+      localStorage.setItem(accessToken, data.accessToken)
+      localStorage.setItem(refreshToken, data.refreshToken)
+      localStorage.setItem(accessExp, data.accessExp)
+      localStorage.setItem(refreshExp, data.refreshExp)
+      router.replace(`${process.env.NEXT_PUBLIC_GAUTH_REDIRECT_URI}`)
+    },
+  })
```

Comment:
```suggestion
  const { fetch } = useFetch<TokensType>({
    url: 'auth',
    method: 'post',
    onSuccess: (data) => {
      if (!data) return
      localStorage.setItem(accessToken, data.accessToken)
      localStorage.setItem(refreshToken, data.refreshToken)
      localStorage.setItem(accessExp, data.accessExp)
      localStorage.setItem(refreshExp, data.refreshExp)
      router.replace(`${process.env.NEXT_PUBLIC_GAUTH_REDIRECT_URI}`)
    },
  })
```

이렇게 사용하라고 generics type 만들어 놨는데

Reply:
main페이지 push하고 이거 고칠려고 할 때 그 사단나서 깜빡했다ㅎ